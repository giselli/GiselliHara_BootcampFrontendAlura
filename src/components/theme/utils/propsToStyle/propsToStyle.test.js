import { propsToStyle } from './index';

describe('propsToStyle()', () => {
  describe('when receive a simple argument', () => {
    test('and it is a string', () => {
      const propsToStyleResult = propsToStyle('textAlign');
      const componentProps = { textAlign: 'center' };
      const styleResult = propsToStyleResult(componentProps);
      expect(styleResult).toEqual({ textAlign: 'center' });
    });
    test('and it is a number', () => {
      const propsToStyleResult = propsToStyle('flex');
      const componentProps = { flex: 1 };
      const styleResult = propsToStyleResult(componentProps);
      expect(styleResult).toEqual({ flex: 1 });
    });
  });
  describe('when receives an argument with breakpoints', () => {
    test('renders only one breakpoints', () => {
      const propsToStyleResult = propsToStyle('textAlign');
      const componentProps = { textAlign: { xs: 'center' } };
      const styleResult = propsToStyleResult(componentProps);
      expect(styleResult).toMatchSnapshot();
    });
    test('renders two or more breakpoints resolutions', () => {
      const propsToStyleResult = propsToStyle('textAlign');
      const componentProps = { textAlign: { xs: 'center', md: 'rigth' } };
      const styleResult = propsToStyleResult(componentProps);
      expect(styleResult).toMatchSnapshot();
    });
  });
});
