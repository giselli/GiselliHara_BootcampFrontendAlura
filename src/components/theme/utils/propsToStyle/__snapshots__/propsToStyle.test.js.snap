// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propsToStyle() when receives an argument with breakpoints renders only one breakpoints 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
]
`;

exports[`propsToStyle() when receives an argument with breakpoints renders two or more breakpoints resolutions 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "text-align: rigth;",
    "}",
  ],
]
`;
